pykafka/utils/compression.py

```diff
@@ -166,3 +187,93 @@ def _detect_xerial_stream(buff):
         header = struct.unpack('!' + _XERIAL_V1_FORMAT, bytes(buff)[:16])
         return header == _XERIAL_V1_HEADER
     return False
+
+
+if lz4:
+    encode_lz4 = lz4.compress  # pylint: disable-msg=no-member
```

Comment:
@sghaskell The tests I wrote are passing just fine without setting the `block_mode` on this call. Am I missing something?

Reply:
I would imagine any simple tests you write will pass without setting block_mode. I found that I had to set it when producing messages due to a lack of support for dependent-block compression.

See [KIP-57](https://cwiki.apache.org/confluence/display/KAFKA/KIP-57+-+Interoperable+LZ4+Framing)

* The only flag left unsupported is dependent-block compression (LZ4 Stream API), which our implementation does not currently support.

Reply:
I wonder how kafka-python manages this https://github.com/dpkp/kafka-python/blob/a3995b38210d3700855d5b438529a5387c095569/kafka/codec.py#L198 cc @dpkp

Reply:
I've actually managed to get this working against both 0.8.2 and 0.10.1 without setting `block_mode`.

Reply:
Awesome! It must have been user error on my part then. Glad to hear it works!